src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacKeyFieldDeclaration/KeyFieldDeclarationAnalyzer.cs

```diff
@@ -0,0 +1,115 @@
+﻿using Acuminator.Utilities.Roslyn;
+using Acuminator.Utilities.Roslyn.PXFieldAttributes;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Acuminator.Analyzers.StaticAnalysis.DacKeyFieldDeclaration
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class KeyFieldDeclarationAnalyzer : PXDiagnosticAnalyzer
+	{
+		private const string IsKey = nameof(PX.Data.PXDBFieldAttribute.IsKey);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create
+			(
+				Descriptors.PX1055_DacKeyFieldsWithIdentityKeyField
+			);
+
+		internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+		{
+			compilationStartContext.RegisterSymbolAction(async symbolContext =>
+				await AnalyzeDacOrDacExtensionDeclarationAsync(symbolContext, pxContext).ConfigureAwait(false), SymbolKind.NamedType);
+		}
+
+		private static async Task AnalyzeDacOrDacExtensionDeclarationAsync(SymbolAnalysisContext symbolContext, PXContext pxContext)
+		{
+			symbolContext.CancellationToken.ThrowIfCancellationRequested();
+
+			if (!(symbolContext.Symbol is INamedTypeSymbol dacOrDacExtSymbol) || !dacOrDacExtSymbol.IsDacOrExtension(pxContext))
+				return;
+
+			var dacPropertiesDeclarations = dacOrDacExtSymbol.GetMembers().OfType<IPropertySymbol>();
+
+			var keyAttributes = new List<AttributeData>();
+
+			bool isKey = false;
+			bool isKeyIdentity = false;
+
+			foreach (var attribute in dacPropertiesDeclarations.SelectMany(a => a.GetAttributes()))
+			{
+				bool hasKeys = attribute.NamedArguments.Any(a => a.Key.Contains(IsKey) &&
+																	a.Value.Value is bool isKeyValue &&
+																	isKeyValue == true);
+
+				if (hasKeys)
+				{
+					var identityOrKey = CheckAttributeIdentityOrKey(attribute, pxContext);
+
+					isKey = identityOrKey.IsKey? true : isKey;
+					isKeyIdentity = identityOrKey.IsKeyIdentity ? true : isKeyIdentity;
```

Comment:
On the line 58 there is no whitespace before `?`, but on the line 59 there is a whitespace before `?`. You should use single code style: the first one or the second one.